packages/core/src/queries/passcode.ts

```diff
@@ -0,0 +1,56 @@
+import { PasscodeType, Passcode, Passcodes, CreatePasscode } from '@logto/schemas';
+import { sql } from 'slonik';
+
+import { buildInsertInto } from '@/database/insert-into';
+import pool from '@/database/pool';
+import { buildUpdateWhere } from '@/database/update-where';
+import { convertToIdentifiers } from '@/database/utils';
+import { DeletionError } from '@/errors/SlonikError';
+
+const { table, fields } = convertToIdentifiers(Passcodes);
+
+export const findUnconsumedPasscodeBySessionIdAndType = async (
```

Comment:
this method is unused, should we keep it?

Reply:
keep if for now, would be used in `verifyPasscode`